public/editor/Sketch.tsx

```diff
@@ -0,0 +1,249 @@
+    };
+    const handleSaturationChange = (newColor: HsvaColor) =>
+      handleChange({ ...hsva, ...newColor, a: hsva.a });
+    const styleMain = {
+      "--sketch-background": "rgb(255, 255, 255)",
+      "--sketch-box-shadow":
+        "rgb(0 0 0 / 15%) 0px 0px 0px 1px, rgb(0 0 0 / 15%) 0px 8px 16px",
+      "--sketch-swatch-box-shadow": "rgb(0 0 0 / 15%) 0px 0px 0px 1px inset",
+      "--sketch-swatch-border-top": "1px solid rgb(238, 238, 238)",
+      background: "var(--sketch-background)",
+      borderRadius: 4,
+      boxShadow: "var(--sketch-box-shadow)",
+      width,
+      ...style,
+    } as CSSProperties;
+    const styleSwatch = {
+      borderTop: "var(--sketch-swatch-border-top)",
+      paddingTop: 10,
+      paddingLeft: 10,
+    } as CSSProperties;
+    const styleSwatchRect = {
+      marginRight: 10,
+      marginBottom: 10,
+      borderRadius: 3,
+      boxShadow: "var(--sketch-swatch-box-shadow)",
+    } as CSSProperties;
+    return (
+      <div
+        {...other}
+        className={`${prefixCls} ${className || ""}`}
+        ref={ref}
+        style={styleMain}
+      >
+        <div style={{ padding: "10px 10px 8px" }}>
+          <Saturation
+            hsva={hsva}
+            style={{ width: "auto", height: 150 }}
+            onChange={handleSaturationChange}
+          />
+          <div style={{ display: "flex", marginTop: 4 }}>
+            <div style={{ flex: 1 }}>
+              <Hue
+                width="auto"
+                height={10}
+                hue={hsva.h}
+                pointer={Bar}
+                innerProps={{
+                  style: { marginLeft: 1, marginRight: 5 },
+                }}
+                onChange={(newHue) => handleChange({ ...hsva, ...newHue })}
+              />
+            </div>
+          </div>
+          {allowAlpha && (
+            <div style={{ display: "flex", marginTop: 4 }}>
+              <div style={{ flex: 1 }}>
+                <Alpha
+                  width="auto"
+                  height={10}
+                  hsva={hsva}
+                  pointer={Bar}
+                  innerProps={{
+                    style: { marginLeft: 1, marginRight: 5 },
+                  }}
+                  onChange={(newHvsa) => {
+                    handleChange({ ...hsva, ...newHvsa });
+                  }}
+                />
+              </div>
+            </div>
+          )}
+        </div>
+        <div style={{ display: "flex", margin: "0 10px 3px 10px" }}>
+          <EditableInput
+            label="Hex"
+            value={hsvaToHex(hsva).replace(/^#/, "").toLocaleUpperCase()}
+            onChange={(_, val) => handleHex(val)}
+            style={{ minWidth: 58 }}
+          />
+          <RGBA
+            hsva={hsva}
+            style={{ marginLeft: 6 }}
+            aProps={allowAlpha ? undefined : false}
+            onChange={(result) => handleChange(result.hsva)}
+          />
+          <EyeDropperButton onChange={handleHex} />
```

Comment:
Vertical alignment of this looks a little strange, maybe it should be closer to the top?